metricbeat/helper/http.go

```diff
@@ -107,8 +109,11 @@ func (h *HTTP) FetchResponse() (*http.Response, error) {
 	}
 
 	req, err := http.NewRequest(h.method, h.uri, reader)
-	if h.base.HostData().User != "" || h.base.HostData().Password != "" {
-		req.SetBasicAuth(h.base.HostData().User, h.base.HostData().Password)
+	if err != nil {
```

Comment:
:-( Sounds like we had a bug before. good catch.

Reply:
Yeah, it panicked on some error scenarios